compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -1449,6 +1449,40 @@
                     "type": "Node"
                 }
             ]
+        },
+        {
+            "name": "ArrayTypeDescriptor",
+            "base": "Node",
+            "kind": "ARRAY_TYPE",
+            "attributes": [
+                {
+                    "name": "typeDescriptorNode",
+                    "type": "Node"
+                },
+                {
+                    "name": "dimensions",
+                    "type": "Node"
```

Comment:
Can we flattened this? i.e: make `[`, `array-len` and `]` part of the ArrayTypeDescriptorNode

Reply:
Yes, since we have changed the structure we can do it now. I'll change it.

Reply:
@SupunS Done please review